mlflow/utils/file_utils.py

```diff
@@ -534,7 +535,13 @@ def local_file_uri_to_path(uri):
     Convert URI to local filesystem path.
     No-op if the uri does not have the expected scheme.
     """
-    path = urllib.parse.urlparse(uri).path if uri.startswith("file:") else uri
+    path = uri
+    if uri.startswith("file:"):
+        parsed_path = urllib.parse.urlparse(uri)
+        path = parsed_path.path
+        # Fix for retaining server name in UNC path.
+        if platform.system().lower() == "windows" and parsed_path.hostname:
```

Comment:
you added a great utility to `test.helper_functions` in the form of `is_local_os_windows` but the same logic is used here. Can that test helper just be moved to MLflow utilities and have the tests import it in the test suites?